src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ILambdaExpression.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    internal partial class IOperationTests : SemanticModelTestBase
+    {
+        [Fact]
+        public void ILambdaExpression_BoundLambda_HasValidLambdaExpressionTree()
+        {
+            string source = @"
+using System;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        Action x /*<bind>*/= () => F()/*</bind>*/;
+    }
+
+    static void F()
+    {
+    }
+}
+";
+            string expectedOperationTree = @"
+IVariableDeclarationStatement (1 declarations) (OperationKind.VariableDeclarationStatement) (Syntax: 'Action x /* ... *</bind>*/;')
+  IVariableDeclaration (1 variables) (OperationKind.VariableDeclaration) (Syntax: 'Action x /* ... *</bind>*/;')
+    Variables: Local_1: System.Action x
+    Initializer: IConversionExpression (ConversionKind.CSharp, Implicit) (OperationKind.ConversionExpression, Type: System.Action) (Syntax: '() => F()')
+        ILambdaExpression (Signature: lambda expression) (OperationKind.LambdaExpression, Type: null) (Syntax: '() => F()')
+          IBlockStatement (2 statements) (OperationKind.BlockStatement) (Syntax: 'F()')
+            IExpressionStatement (OperationKind.ExpressionStatement) (Syntax: 'F()')
+              IInvocationExpression (static void Program.F()) (OperationKind.InvocationExpression, Type: System.Void) (Syntax: 'F()')
+            IReturnStatement (OperationKind.ReturnStatement) (Syntax: 'F()')
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyOperationTreeAndDiagnosticsForTest<EqualsValueClauseSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact]
+        public void ILambdaExpression_UnboundLambdaAsVar_HasValidLambdaExpressionTree()
+        {
+            string source = @"
+using System;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        var x /*<bind>*/= () => F()/*</bind>*/;
+    }
+
+    static void F()
+    {
+    }
+}
+";
+            string expectedOperationTree = @"
+IVariableDeclarationStatement (1 declarations) (OperationKind.VariableDeclarationStatement, IsInvalid) (Syntax: 'var x /*<bi ... *</bind>*/;')
+  IVariableDeclaration (1 variables) (OperationKind.VariableDeclaration) (Syntax: 'var x /*<bi ... *</bind>*/;')
+    Variables: Local_1: var x
+    Initializer: ILambdaExpression (Signature: lambda expression) (OperationKind.LambdaExpression, Type: null) (Syntax: '() => F()')
```

Comment:
A note here: a conversion expression is not generated here, as the type could not be deduced. In all other aspects a conversion expression is expected above the lambda expression. We even have one in VB for the similar code:
```VB
Dim a = Sub()
        End Sub()
```
```
IOperation tree for "Dim a = Sub()

                End Sub"

IVariableDeclarationStatement (1 declarations) (OperationKind.VariableDeclarationStatement) (Syntax: 'Dim a = Sub ... End Sub')
  IVariableDeclaration (1 variables) (OperationKind.VariableDeclaration) (Syntax: 'a')
    Variables: Local_1: a As Sub <generated method>()
    Initializer: IConversionExpression (ConversionKind.Basic, Implicit) (OperationKind.ConversionExpression, Type: Sub <generated method>()) (Syntax: 'Sub() ... End Sub')
        ILambdaExpression (Signature: Sub ()) (OperationKind.LambdaExpression, Type: null) (Syntax: 'Sub() ... End Sub')
          IBlockStatement (2 statements) (OperationKind.BlockStatement) (Syntax: 'Sub() ... End Sub')
            ILabelStatement (Label: exit) (OperationKind.LabelStatement) (Syntax: 'End Sub')
            IReturnStatement (OperationKind.ReturnStatement) (Syntax: 'End Sub')
```
Do we think there should be a conversion expression here? @dotnet/analyzer-ioperation @CyrusNajmabadi @AlekseyTs.